src/EditorFeatures/CSharpTest/GenerateVariable/GenerateVariableTests.cs

```diff
@@ -2511,6 +2511,17 @@ public static int ExMethod(this string s)
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
+        public async Task SpeakableTopLevelStatementType()
+        {
+            await TestMissingAsync(@"
+[|P|] = 10;
+
+partial class Program
+{
+}");
```

Comment:
A more ideal behavior is to generate the property as follows:

```csharp
P = 10;

partial class Program
{
    public int P { get; private set; }
}
```

In case you find it out-of-scope for this PR, we can open a tracking issue for that.